sdk/monitor/azure-monitor-query/azure/monitor/query/_metrics_batch_query_client.py

```diff
@@ -0,0 +1,143 @@
+#
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from json import loads
+from typing import Any, List, MutableMapping
+
+from azure.core.credentials import TokenCredential
+from azure.core.tracing.decorator import distributed_trace
+
+from ._generated.metrics.batch import MonitorBatchMetricsClient
+from ._models import MetricsQueryResult
+from ._helpers import get_authentication_policy, get_timespan_iso8601_endpoints, get_subscription_id_from_resource
+
+JSON = MutableMapping[str, Any]  # pylint: disable=unsubscriptable-object
+
+
+class MetricsBatchQueryClient(object):  # pylint: disable=client-accepts-api-version-keyword
+    """MetricsBatchQueryClient should be used for performing metrics queries on multiple monitored resoures in the same
+    region. A credential with authorization at the subscription level is required when using this client.
+
+    :param credential: The credential to authenticate the client.
+    :type credential: ~azure.core.credentials.TokenCredential
+    :param str endpoint: The regional endpoint to use, for example
+        https://eastus.metrics.monitor.azure.com. The region should match the region of the requested
+        resources. For global resources, the region should be 'global'. Required.
+    :keyword str audience: The audience to use when requesting a token. If not provided, the public cloud audience
+        will be assumed. Defaults to 'https://metrics.monitor.azure.com'.
+    """
+
+    def __init__(self, credential: TokenCredential, endpoint: str, **kwargs: Any) -> None:
+        self._endpoint = endpoint
+        if not self._endpoint.startswith("https://") and not self._endpoint.startswith("http://"):
+            self._endpoint = "https://" + self._endpoint
+        audience = kwargs.pop("audience", "https://metrics.monitor.azure.com")
+
+        authentication_policy = kwargs.pop("authentication_policy", None) or get_authentication_policy(
+            credential, audience
+        )
+
+        self._client = MonitorBatchMetricsClient(
+            credential=credential, endpoint=self._endpoint, authentication_policy=authentication_policy, **kwargs
+        )
+        self._batch_metrics_op = self._client.metrics
+
+    @distributed_trace
+    def query_batch(
+        self, resource_uris: List[str], metric_namespace: str, metric_names: List[str], **kwargs: Any
+    ) -> List[MetricsQueryResult]:
```

Comment:
I think it might be good to check with architects + the service team if they ever expect this might need paging. For return collections like this, sometimes in the SDK we'll implement it as an ItemPaged so that it won't be breaking if the service decides to support paging later on.

Here's the applicable guideline: https://azure.github.io/azure-sdk/python_design.html#python-paged-non-server-paged-list

Reply:
Thanks for the guideline link, and that's a great point. I will double check this before GA of this client in a couple months. For now, since this is only for preview, and other language SDKs already shipped with the result as non-pageable, I will keep this as is.